pkg/providers/vsphere/vsphere.go

```diff
@@ -538,6 +538,9 @@ func NeedsNewWorkloadTemplate(oldSpec, newSpec *cluster.Spec, oldVdc, newVdc *v1
 	if oldSpec.Bundles.Spec.Number != newSpec.Bundles.Spec.Number {
 		return true
 	}
+	if !v1alpha1.WorkerNodeGroupConfigurationSliceTaintsEqual(oldSpec.Spec.WorkerNodeGroupConfigurations, newSpec.Spec.WorkerNodeGroupConfigurations) {
```

Comment:
Do we need this for docker or other providers too?

Reply:
yeah, it'll be added to other providers; I actually just pushed that a few minutes ago after testing it out on vsphere and confirming the behavior (e.g. it causes an upgrade to happen and the nodes to roll, which it does)

Reply:
done for docker